File: frontend-moviez-app/src/app/shared/allotment-service/allotment.service.ts
import { Injectable } from '@angular/core';
import { Allotment } from 'src/app/movie-multiplex-allotment/model/allotment';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class AllotmentService {

  baseUrl = "http://localhost:8765/allotment-service/allot/";

  constructor(private http: HttpClient) { }

  saveAllotmentDetails(allotment: Allotment): Promise<Allotment> {
    return this.http.post(this.baseUrl+'add', allotment).toPromise()
      .then(response => response as Allotment);
  }

  getAllotmentDetails(multiplexName: String): Promise<Array<Allotment>> {
    return this.http.get(this.baseUrl+  `get/` + multiplexName)
      .toPromise()
      .then(response => response as Allotment[])
  }

  getScreenNumberForMultiplex(multiplexName: String): Promise<Array<Allotment>> {
    return this.http.get(this.baseUrl + 'getScreen/' + multiplexName)
      .toPromise()
      .then(response => response as Allotment[])
  }

}

File: backend-moviez-app/allotment-service/src/main/java/com/application/allotment/service/AllotmentService.java
package com.application.allotment.service;

import java.util.List;

import com.application.allotment.document.Multiplex;
import com.application.allotment.dto.AllotmentDto;

public interface AllotmentService {

	public AllotmentDto saveAllotmentDetails(AllotmentDto dto);

	public List<AllotmentDto> getAllotmentDetails();

//	public List<AllotmentDto> getScreenNumbersOccupiedByMultiplex(String multiplexName);
	
	public Multiplex getScreenNumbersOccupiedByMultiplex(String multiplexName);
}
